### DISCUSSION TITLE
Disable the `unicorn/prefer-module` rule

### DISCUSSION THREAD

**Discussion by leroykorterink** [mediamonks/eslint-config#231] [General]
There is a proposal to disable the `unicorn/prefer-module` rule.

> `unicorn/prefer-module`: this breaks inline require(), which can be really useful in some cases.
> \- @edgardz

**Comment by leroykorterink** [comment]
Prefer using the JavaScript module format over the legacy CommonJS module format. Together with other changes, this helps the ecosystem migrate to a single, native module format.

It's fine to disable the rule for a single line if you **must** use the `require` keyword in a project.

Please see [the documentation](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-module.md) of the rule.

**Reply by ThaNarie** [reply]
I think this is mostly a configuration issue when writing code for `node` instead of the browser.

- If you write ESM, you should use this rule, as `require` doesn't work in that context.
- If you write CJS, you should NOT use this rule, as you need `require`
- If you're writing frontend code, also use `import` since all bundlers work better with that.

The current issue is that frontend projects might also have `scripts` in them that are `cjs`, or next api middleware that still uses `cjs`.

In those scenario it should be easy to switch to a `node-cjs` eslint configuration for those files/folders, that doesn't have this rule enabled.
